plugins/destination/gcs/client/spec/schema.go

```diff
@@ -39,38 +39,46 @@ func (s Spec) JSONSchemaExtend(sc *jsonschema.Schema) {
 				return properties
 			}(),
 		},
+		Extras: map[string]interface{}{
+			"errorMessage": map[string]interface{}{
+				"properties": map[string]interface{}{
+					"no_rotate":        "batching options must not be present when no_rotate is enabled",
+					"batch_size":       "batching options must not be present when no_rotate is enabled",
+					"batch_size_bytes": "batching options must not be present when no_rotate is enabled",
+					"batch_timeout":    "batching options must not be present when no_rotate is enabled",
+				},
+			},
+		},
 	}
 
 	// path patterns: should be a clean path
 	cleanPath := &jsonschema.Schema{
 		Title: "`path` is a clean path value",
-		Not: &jsonschema.Schema{
-			Title: "`path` is not a clean path value",
-			AnyOf: []*jsonschema.Schema{
-				{
-					Title: "`path` contains `./`",
-					Properties: func() *orderedmap.OrderedMap[string, *jsonschema.Schema] {
-						properties := orderedmap.New[string, *jsonschema.Schema]()
-						properties.Set("path", &jsonschema.Schema{
-							Type:    "string",
-							Pattern: `^.*\./.*$`,
-						})
-						return properties
-					}(),
-				},
-				{
-					Title: "`path` contains `//`",
-					Properties: func() *orderedmap.OrderedMap[string, *jsonschema.Schema] {
-						properties := orderedmap.New[string, *jsonschema.Schema]()
-						properties.Set("path", &jsonschema.Schema{
-							Type:    "string",
-							Pattern: `^.*//.*$`,
-						})
-						return properties
-					}(),
+		Extras: map[string]interface{}{
```

Comment:
Simplify the schema and also allow using `errorMessage->properties->path` mapping